backend/server.go

```diff
@@ -0,0 +1,183 @@
+	err := tmpl.ExecuteTemplate(w, "index.html", Data{
+		People: people,
+		Awards: awards,
+	})
+	checkError(w, err)
+}
+
+func checkError(w http.ResponseWriter, err error) {
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+func (app *App) homeHandler(w http.ResponseWriter, r *http.Request) {
+	people, err := app.loadPeople()
+	checkError(w, err)
+
+	renderTemplate(w, people, app.tmpl)
+
+	awards = make([]string ,0)
+}
+
+func (app *App) addHandler(w http.ResponseWriter, r *http.Request) {
+	conn := app.pool.Get()
+
+	err := r.ParseForm()
+	checkError(w, err)
+
+	for key, values := range r.Form {
+		name := strings.SplitN(key, "-", 2)[1]
+		value := values[0]
+
+		if value == "" {
+			continue
+		}
+
+		byPoints, err := strconv.ParseFloat(value, 64)
+		checkError(w, err)
+
+		// check awards
+		currScore, err := redis.Float64(conn.Do("HGET", name, "score"))
+		checkError(w, err)
+		if currScore + byPoints >= 100 {
+			awards = append(awards, name)
+			byPoints -= 100
+		}
+		_, err = conn.Do("HINCRBYFLOAT", name, "score", byPoints)
+		checkError(w, err)
+	}
+	http.Redirect(w, r, "/", http.StatusFound)
+}
+
+func (app *App) addPeopleHandler(w http.ResponseWriter, r *http.Request){
+	conn := app.pool.Get()
+
+	name, score, expression := r.FormValue("new-name"), r.FormValue("new-score"), r.FormValue("new-score-exp")
+
+	existed, err := redis.Bool(conn.Do("EXISTS", name))
+	checkError(w, err)
+
+	if !existed {
+		// check award
+		scoreVal, err := strconv.ParseFloat(score, 64)
+		if scoreVal >= 100 {
+			awards = append(awards, name)
+			scoreVal -= 100
+		}
+		_, err = conn.Do("HSET", name, "name", name, "score", scoreVal, "expression", expression)
+		checkError(w, err)
+	}
+
+	http.Redirect(w, r, "/", http.StatusFound)
+}
+
+func (app *App) deletePeopleHandler(w http.ResponseWriter, r *http.Request) {
+	conn := app.pool.Get()
+
+	err := r.ParseForm()
+	checkError(w, err)
+
+	for key, _ := range r.Form {
+		_, err := conn.Do("DEL", key)
+		checkError(w, err)
+	}
+
+	http.Redirect(w, r, "/", http.StatusFound)
+}
+
+func (app *App) loadPeople() ([]Person, error) {
+	conn := app.pool.Get()
+	defer conn.Close()
+
+	peopleKeys, err := redis.Strings(conn.Do("KEYS", "*"))
+	if err != nil {
+		log.Fatal(err)
+		return nil, err
+	}
+	sort.Strings(peopleKeys)
+
+	people := make([]Person, len(peopleKeys))
+	for idx, personName := range peopleKeys {
+		if personName == ""{
+			conn.Do("DEL", personName)
+		}
+		raw, err := redis.Values(conn.Do("HGETALL", personName))
+		person := Person{}
+		err = redis.ScanStruct(raw, &person)
+		if err != nil {
+			continue
+		}
+		people[idx] = person
+	}
+	return people, nil
+}
+
+func (app* App) init(mode string) {
+	switch  mode {
+	case "DEVELOPMENT":
+		app.config = devConfig
+	case "TESTING":
+		app.config = testConfig
+	case "DEPLOYMENT":
+		app.config = deployConfig
+	default:
+		app.config = devConfig
+	}
```

Comment:
You might want to restructure this as the following since both development and default have the same config:
```
case "TESTING":
    	app.config = testConfig
case "DEPLOYMENT":
	app.config = deployConfig
case "DEVELOPMENT":
default:
	app.config = devConfig
```